data_steward/validation/main.py

```diff
@@ -1037,34 +1040,70 @@ def union_ehr():
 @api_util.auth_required_cron
 @log_traceback
 def run_retraction_cron():
+    """
+    Run a cron job to mimic the run_retraction.py script
+    """
     project_id = bq_utils.app_identity.get_application_id()
     hpo_id = bq_utils.get_retraction_hpo_id()
     retraction_type = bq_utils.get_retraction_type()
     pid_table_id = bq_utils.get_retraction_pid_table_id()
     sandbox_dataset_id = bq_utils.get_retraction_sandbox_dataset_id()
 
+    # Dataset and table containing list of datasets
+    datasets_to_retract_dataset = bq_utils.get_retraction_dataset_ids_dataset()
+    datasets_to_retract_table = bq_utils.get_retraction_dataset_ids_table()
+
     # retract from bq
-    dataset_ids = bq_utils.get_retraction_dataset_ids()
-    logging.info(f"Dataset id/s to target from env variable: {dataset_ids}")
+    if not datasets_to_retract_table or not datasets_to_retract_dataset:
+        logging.info(
+            f"Retraction cannot run without RETRACTION_DATASET_IDS_TABLE and RETRACTION_DATASET_IDS_DATASET"
+        )
+        return 'retraction-skipped'
+
+    bq_client = BigQueryClient(project_id)
+    dataset_query_job = bq_client.query(
+        f"SELECT * FROM {project_id}.{datasets_to_retract_dataset}.{datasets_to_retract_table}"
+    )
+    dataset_ids_result = dataset_query_job.result()
+    dataset_ids = dataset_ids_result.to_dataframe()["tables"].to_list()
+    logging.info(f"Dataset id/s to target from table: {dataset_ids}")
     logging.info(f"Running retraction on BQ datasets")
 
     # retract from default dataset
-    retract_data_bq.run_bq_retraction(project_id, sandbox_dataset_id,
-                                      pid_table_id, hpo_id, dataset_ids,
-                                      retraction_type)
+    retract_data_bq.run_bq_retraction(project_id,
+                                      sandbox_dataset_id,
+                                      pid_table_id,
+                                      hpo_id,
+                                      dataset_ids,
+                                      retraction_type,
+                                      bq_client=bq_client)
     logging.info(f"Completed retraction on BQ datasets")
 
-    # retract from gcs
-    folder = bq_utils.get_retraction_submission_folder()
-    logging.info(f"Submission folder/s to target from env variable: {folder}")
-    logging.info(f"Running retraction from internal bucket folders")
-    retract_data_gcs.run_gcs_retraction(project_id,
-                                        sandbox_dataset_id,
-                                        pid_table_id,
-                                        hpo_id,
-                                        folder,
-                                        force_flag=True)
-    logging.info(f"Completed retraction from internal bucket folders")
+    # Run memory changes
+    for dataset_id in dataset_ids:
+        logging.info(f"Running CRs for {dataset_id}...")
+        cleaning_args = [
```

Comment:
`DATA_CONSISTENCY` has these 3 cleaning rules, and `1` and `2` fail with some old datasets:
1. `DropOrphanedSurveyConductIds`
2. `DropOrphanedPIDS`
3. `CleanMappingExtTables`

`survey_conduct` exits only after `2022q4`. So, `1` needs to run on datasets from `2022q4` and newer.
`2` looks at `visit_detail`, and `visit_detail` exists only  after `2022q2`. So, `2` needs to run on datasets from `2022q2` and newer.
`3` should run with no problem with any datasets.

https://github.com/all-of-us/curation/blob/develop/data_steward/cdr_cleaner/clean_cdr.py#L349

So, instead of running `DATA_CONSISTENCY` for everything, we might need to run individual cleaning rules depending on which dataset it is.

Reply:
Thanks, I added a `RETRACTION_CRON` cleaning stage with only the ext/mapping cleaning rule.